p4-16/spec/P4-16-spec.mdk

```diff
@@ -2642,6 +2642,7 @@ P4 supports extern object declarations and extern function declarations using th
 externDeclaration
     : optAnnotations EXTERN nonTypeName optTypeParameters '{' methodPrototypes '}'
     | optAnnotations EXTERN functionPrototype ';'
+    | optAnnotations EXTERN name ';'
```

Comment:
@jafingerhut Action item for me to propose adding a sentence or two to the spec for this.  The basic idea is that it enables giving a kind of forward declaration for an extern name, before it is fully defined.  Chris Dodd believes the motivation for this being in the grammar was for Register and RegisterAction definition in TNA.

Reply:
I have added some proposed text for this, which should be reviewed.

Reply:
During 2021-May-03 LDWG meeting it was decided to leave forward extern declarations out of the spec.  It will remain in the p4c implementation, which already includes several features marked experimental that are not in the spec.

AI Andy: Remove this part of this PR and update it today.